src/ParameterLists.f90

```diff
@@ -4718,12 +4718,14 @@ SUBROUTINE edit_ParamType_STR(thisParam,funit,indent,prefix,paddtw)
         ENDIF
       ENDIF
       WRITE(fmt,'(i12)') i; fmt=ADJUSTL(fmt)
+      tmpstr=''
+      IF(LEN_TRIM(thisParam%val) > 0) tmpstr=thisParam%val
       IF(LEN_TRIM(thisParam%description) == 0) THEN
         WRITE(UNIT=funit,FMT='('//TRIM(fmt)//'x,a)') sprefix// &
-          sdtype//' :: '//thisParam%name//'='//thisParam%val
+          sdtype//' :: '//thisParam%name//'='//tmpstr
```

Comment:
why write ```tmpstr```?

Just write ```sdtype//' :: '//thisParam%name//'='```

Reply:
I didn't want to have a nested set of IF-blocks.  Because we already check if the description is LEN_TRIM=0, now you have 4 logical cases, and it looks ugly.  I can simplify it differently if you'd prefer.  I may do it anyways to get rid of the IF-ELSE block and just have a single write statement.